packages/router/src/lib/helpers/shared.ts

```diff
@@ -152,6 +192,50 @@ export const calculateGasFeeInReceivingTokenForFulfill = async (
   );
 };
 
+/**
+ * Helper to calculate gas fee for specific transactions
+ *
+ * @param chainId The network id that we're going to get gas fee on
+ * @param assetId The asset address that we're going to get gas fee in
+ * @param decimals Decimal number of assets
+ * @param method "prepare" | "fulfill" | "cancel"
+ * @param requestContext Request context instance.
+ * @param methodContext Method context instance.
+ * @param whichChain "sending" | "receiving"
+ */
+export const calculateGasFee = async (
+  chainId: number,
+  assetId: string,
+  decimals: number,
+  method: "prepare" | "fulfill" | "cancel",
+  requestContext: RequestContext,
+  methodContext: MethodContext,
+): Promise<BigNumber> => {
+  const { txService } = getContext();
+
+  const assetIdOnMainnet = await getMainnetEquivalent(assetId, chainId);
+  const tokenPricingChain = assetIdOnMainnet ? 1 : chainId;
+  const tokenPricingAssetId = assetIdOnMainnet ? assetIdOnMainnet : assetId;
+
+  const nativeAssetIdOnMainnet = await getMainnetEquivalent(constants.AddressZero, chainId);
+  const nativeTokenPricingChain = nativeAssetIdOnMainnet ? 1 : chainId;
+  const nativeTokenPricingAssetId = nativeAssetIdOnMainnet ? nativeAssetIdOnMainnet : constants.AddressZero;
+
+  const gasFeeRes = await txService.calculateGasFee(
```

Comment:
tx service will also need to know if they should account for bump in estimates due to transaction manager vs. router contract calls (these should have different estimates)

Reply:
Good call this is true. cc @sanchaymittal